core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -194,7 +194,7 @@ abstract class RDD[T: ClassTag](
   @transient private var partitions_ : Array[Partition] = null
 
   /** An Option holding our checkpoint RDD, if we are checkpointed */
-  private def checkpointRDD: Option[RDD[T]] = checkpointData.flatMap(_.checkpointRDD)
+  private def checkpointRDD: Option[CheckpointRDD[T]] = checkpointData.flatMap(_.checkpointRDD)
```

Comment:
how is this hardcoding? Semantically we want the type of this to always be `CheckpointRDD`. This is just enforcing it in code.